injection/sharedmain/main.go

```diff
@@ -220,3 +225,16 @@ func flush(logger *zap.SugaredLogger) {
 	logger.Sync()
 	metrics.FlushExporter()
 }
+
+// Components should match this regex from the Prometheus documentation
+// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+const componentNameRegex = "^[a-zA-Z_:]([a-zA-Z0-9_:])*$"
+
+var (
+	componentNameValiditor = regexp.MustCompile(componentNameRegex)
+)
+
+// isComponentNameValid returns true if valid, false otherwise
+func isComponentNameValid(name string) bool {
+	return componentNameValiditor.MatchString(name)
```

Comment:
At this point I am not sure that this function makes sense. Just inline it.

Reply:
I moved the functionality to a function so I could test it in isolation. I can inline it but I don't think the tests could and wold be so explicit.

Reply:
You can just check regex matches strings you expect in the test...

Reply:
@vagababov `MainWithConfig()` is a really long function, it's not a great place to test a regexp IMO.